core/src/consensus/fork_choice.rs

```diff
@@ -64,31 +103,7 @@ fn select_forks_failed_switch_threshold(
     total_stake: u64,
     switch_fork_decision: SwitchForkDecision,
 ) -> SwitchForkDecision {
-    let last_vote_unable_to_land = match reset_bank {
-        Some(heaviest_bank_on_same_voted_fork) => {
-            match tower.last_voted_slot() {
-                Some(last_voted_slot) => {
-                    match progress.my_latest_landed_vote(heaviest_bank_on_same_voted_fork.slot()) {
-                        Some(my_latest_landed_vote) =>
-                        // Last vote did not land
-                        {
-                            my_latest_landed_vote < last_voted_slot
-                                // If we are already voting at the tip, there is nothing we can do.
-                                && last_voted_slot < heaviest_bank_on_same_voted_fork.slot()
-                                // Last vote outside slot hashes of the tip of fork
-                                && !heaviest_bank_on_same_voted_fork
-                                    .is_in_slot_hashes_history(&last_voted_slot)
-                        }
-                        None => false,
-                    }
-                }
-                None => false,
-            }
-        }
-        None => false,
-    };
-
-    if last_vote_unable_to_land {
+    if !last_vote_able_to_land(reset_bank, progress, tower) {
```

Comment:
collapsed this down and early return (instead of if/else) to reduce indent for the more complex else leg